In [56]:

import re
import datetime
import pandas as pd

log_path = "setupapi.dev2.log"
usb_devices = []
# Read the contents of the setupapi.dev.log file
with open(log_path, "r") as log_file:
    # Store information about each USB device in a dictionary
    for line in log_file:
        match = re.match(r'^>>>  \[Device Install.*#(Disk&Ven_[A-Za-z0-9]+)&(Prod_([\w\s\S]+?))&(Rev_([\w\s\S]+?))#([\w\s\S]+?)#.*\]', line)
        if match:
            usb_devices.append({
                'device_vendor_id': match.group(1),
                'device_product_id': match.group(2),
                'device_instance_id': match.group(4),
                'serial_number': match.group(6),
                'event_time': next(log_file).split("start")[1].split("\n")[0],
            })

df = pd.DataFrame(usb_devices)

display(df)


device_vendor_id      device_product_id device_instance_id  \
0  Disk&Ven_Verbatim        Prod_STORE_N_GO           Rev_1100   
1       Disk&Ven_SMI          Prod_USB_DISK           Rev_1100   
2  Disk&Ven_VendorCo       Prod_ProductCode           Rev_2.00   
3  Disk&Ven_VendorCo       Prod_ProductCode           Rev_2.00   
4  Disk&Ven_Kingston  Prod_DataTraveler_2.0           Rev_PMAP   
5  Disk&Ven_Kingston  Prod_DataTraveler_2.0           Rev_PMAP   
6  Disk&Ven_JetFlash        Prod_TS128MJF2A           Rev_1.00   
7  Disk&Ven_JetFlash        Prod_TS128MJF2A           Rev_1.00   

                serial_number                event_time  
0          1311130000000480&0   2022/11/03 15:00:17.389  
1                 6&7d7590d&0   2023/02/01 11:10:46.945  
2       920707402A3F8066956&0   2023/02/06 12:50:42.455  
3       9207113E5CF83172266&0   2023/02/06 12:51:00.660  
4  AC220B280549EF6059EEBAFE&0   2023/02/06 12:52:25.253  
5  AC220B280549EF6059EEBAFE&0   2023/03/23 08:13:57.618  
6                6&22ff249f&1   2023/03/29 16:28:31.704  
7                6&104573fa&1   2023/03/29 16:32:50.762

In [57]:
import os
import pandas as pd
import hashlib
import time

dir_path = "D:\\"

# f = magic.Magic()

file_names = []
original_extensions = []
extension_matches = []
md5s = []
sha1s = []
sha256s = []
magic_numbers = []
creation_time = []
modification_time = []
access_time = []


for file in os.listdir(dir_path):
    if os.path.isfile(os.path.join(dir_path, file)):
        print(file)
        file_name, extension = os.path.splitext(file)
        file_names.append(file_name)
        original_extensions.append(extension)
        md5s.append(hashlib.md5(file.encode('utf-8')).hexdigest())
        sha1s.append(hashlib.sha1(file.encode('utf-8')).hexdigest())
        sha256s.append(hashlib.sha256(file.encode('utf-8')).hexdigest())
        creation_time.append(time.ctime(os.path.getctime(os.path.join(dir_path,file))))
        modification_time.append(time.ctime(os.path.getmtime(os.path.join(dir_path,file))))
        access_time.append(time.ctime(os.path.getatime(os.path.join(dir_path,file))))

df = pd.DataFrame({
    'file_name': file_names,
    'extension': original_extensions,
    'md5': md5s,
    'sha1': sha1s,
    'sha256': sha256s,
    })

df2 = pd.DataFrame({
    "creation_time": creation_time,
    "modification_time": modification_time,
    'access_time': access_time,
})

display(df)
display(df2)

Secret_file_11.txt
Secret_file_92.pdf


file_name extension                               md5  \
0  Secret_file_11      .txt  3b40bc1a04ff7d4038b2b156bf65d6fc   
1  Secret_file_92      .pdf  0ffa4083c83eb44465a1d8cfe35e4bc9   

                                       sha1  \
0  14c5a20a75a2e94747261cb2105d41d7f8367e15   
1  012a03542c816c6fe2883c15c104aef41b6c834b   

                                              sha256  
0  60ce7243ad35a0a9795d15d561d56e3f20f67ec4c6d0dd...  
1  b54385ca9f043d3566413bd00868ca433c5ed89de9960d...

creation_time         modification_time  \
0  Wed Mar 29 17:35:08 2023  Mon Oct 28 16:08:46 2019   
1  Wed Mar 29 17:35:10 2023  Mon Oct 28 16:09:32 2019   

                access_time  
0  Wed Mar 29 01:00:00 2023  
1  Wed Mar 29 01:00:00 2023

Disk&Ven_JetFlash	Prod_TS128MJF2A	Rev_1.00	6&104573fa&1	2023/03/29 16:32:50.762 je rijesenje zadatka